src/trunk.jl

```diff
@@ -338,29 +369,23 @@ function solve!(
     update!(tr, sNorm)
 
     optimal = ∇fNorm2 ≤ ϵ
-    elapsed_time = time() - start_time
-    tired = neval_obj(nlp) > max_eval ≥ 0 || elapsed_time > max_time
-    solved = optimal || tired || stalled
-  end
-  verbose > 0 && @info log_row(Any[iter, f, ∇fNorm2, tr.radius])
-
-  if optimal
-    status = :first_order
-  elseif tired
-    if neval_obj(nlp) > max_eval ≥ 0
-      status = :max_eval
-    elseif elapsed_time > max_time
-      status = :max_time
-    end
+    output.elapsed_time = time() - start_time
+
+    output.status = get_status(
+      nlp,
+      elapsed_time = output.elapsed_time,
+      optimal = optimal,
+      max_eval = max_eval,
+      max_time = max_time,
+    )
+
+    callback(nlp, solver)
+
+    done = output.status != :unknown
   end
+  verbose > 0 && @info log_row(Any[output.iter, output.objective, ∇fNorm2, tr.radius])
 
-  return GenericExecutionStats(
-    status,
-    nlp,
-    solution = x,
-    objective = f,
-    dual_feas = ∇fNorm2,
-    iter = iter,
-    elapsed_time = elapsed_time,
-  )
+  output.dual_feas = ∇fNorm2
```

Comment:
I think this could also be done during the loop, similar to `output.objective`.

Reply:
Done. It almost makes sense to change `get_status` to compute `optimal`, `tired` and etc.

Reply:
Something like `status = get_status(nlp, output, criteria)`. Not for this PR, though.